### PR TITLE
Fields location in config

### PR DESCRIPTION
**Pull request by johnollier**
NOT for merging until new config items have been added for test etc. Just for review.

I've added the ability to read classpath://xxx urls and then set the default config items in the project to be classpath urls. This is to reduce the number of Optional objects being passed around when the config is read as the application will complain and fail to start if the config items are missing. I also tidied up a couple of tests.

### CODE REVIEW

File: src/main/java/uk/gov/register/configuration/ConfigManager.java [hunk 1]
```diff
@@ -57,11 +52,11 @@ public FieldsConfiguration getFieldsConfiguration() {
         return fieldsConfiguration.get();
     }
 
-    private RegistersConfiguration createRegistersConfiguration() {
-        return new RegistersConfiguration(registersConfigFileUrl.map(s -> registersConfigFilePath));
+    private RegistersConfiguration createRegistersConfiguration() throws IOException {
```
Comment (karlbaker02) [inline src/main/java/uk/gov/register/configuration/ConfigManager.java]: Should this throw `IOException` when the corresponding `createFieldsConfiguration` method doesn't?

File: src/main/java/uk/gov/register/protocols/classpath/ClasspathURLConnection.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ClasspathURLConnection extends URLConnection {
+
+    private final Pattern classpathPattern = Pattern.compile("classpath://(.+)");
+
+    ClasspathURLConnection(URL url) {
```
Comment (karlbaker02) [inline src/main/java/uk/gov/register/protocols/classpath/ClasspathURLConnection.java]: Protected constructor?

Comment (johnollier) [inline src/main/java/uk/gov/register/protocols/classpath/ClasspathURLConnection.java]: It only gets called from uk.gov...protocols.classpath.Handler

File: src/main/java/uk/gov/register/configuration/ConfigManager.java (hunk 1 above)
Comment (johnollier) [inline src/main/java/uk/gov/register/configuration/ConfigManager.java]: You're right.

**Comment by chr1sdi** [comment]
👍

**Pull request merged**